google_maps/public/js/src/google_maps_edit.js

```diff
@@ -76,8 +130,41 @@ function GoogleMapsEditXBlock(runtime, element) {
   $(element).on('click', '.clear-iframe-height', function() {
     setDefaultValue(iframeHeightInput, iframeHeightDefaultValue, iframeHeightClearButton);
   });
-  
-  $(element).on('click', '.save-button', saveXBlockData);
+
+  $(element).on('click', '.save-button', validatePlaceNameField);
 
   $(element).on('click', '.cancel-button', exitEditMode);
+
+  $(element).on('blur', 'input[id=api-key-input]', function() {
+    googleMapsXBlockApp.googleMapsApiKey = $('input[id=api-key-input]').val();
+  });
 }
+
+googleMapsXBlockApp.initMap = function () {
+  var placeNameInput = $('#place-name-input'),
+      autocomplete = new google.maps.places.Autocomplete(placeNameInput[0]),
+      getParameters = {};
+
+  autocomplete.addListener('place_changed', function () {
+    var place = autocomplete.getPlace();
+
+    if (!place.geometry) {
+      // User entered the name of a Place that was not suggested and
+      // pressed the Enter key or the Place Details request failed.
+      googleMapsXBlockApp.toggleErrorMessage(placeNameInput, true);
+      return;
+    }
+
+    googleMapsXBlockApp.toggleErrorMessage(placeNameInput, false);
+    googleMapsXBlockApp.place = {
+      'id': place.place_id,
+      'name': placeNameInput.val()
+    };
+    getParameters = {
+      key: googleMapsXBlockApp.googleMapsApiKey,
+      q: 'place_id:' + place.place_id
+    };
+
+    googleMapsXBlockApp.changeEmbeddedMapLocation(getParameters);
+  });
+};
```

Comment:
Calling API can take some time if the internet connection of the end user is poor.

1) Add a spinner to give staff members a sense of something happening in the background as the result of their interaction.
2) Save button should be disabled during this process.